lib/posix-socket/socket.c

```diff
@@ -341,9 +341,14 @@ int uk_sys_accept(const struct uk_file *sock, int blocking,
 	_socket_init(al, n->driver, new_data);
 
 	al_listener = __containerof(sock, struct socket_alloc, f);
+#if CONFIG_LIBPOSIX_SOCKET_EVENTS
 	uk_socket_evd_init_from(&al->evd, &al_listener->evd);
 	uk_socket_evd_laddr_set_from(&al->evd, &al_listener->evd);
-	uk_socket_evd_raddr_set(&al->evd, addr, *addr_len);
+	if (addr_len)
+		uk_socket_evd_raddr_set(&al->evd, addr, *addr_len);
+	else
+		al->evd.raddr_len = 0;
```

Comment:
I think it's more appropriate for the `NULL` check to be done inside `uk_socket_evd_raddr_set()`, as that function already makes a provision for that case:
```diff
-        if (!raddr || (raddr_len == 0)) {
+        if (!raddr || !raddr_len || (raddr_len == 0)) {
                /* Special case: we unset the address */
                evd->raddr_len = 0;
                return;
        }
```

Reply:
Unfortunately we can't check raddr_len in uk_socket_evd_raddr_set without changing function interface. That's because we pass *addr_len which raises null pointer on the function call

Reply:
Yes, that's what I actually meant as `socket.c` seems to treat `struct uk_socket_event_data` as an opaque struct. I just forgot to add the indirection in the check against zero:
```
        if (!raddr || !raddr_len || (*raddr_len == 0)) {
```
Let's also see what @skuenzer thinks about updating the interface as he's the original author of that API.

Reply:
Hey, I'd prefer if you use that function so that the fields of the `struct uk_socket_event_data evd;` is not directly used in this file for code separation reasons.
However, you could invoke the function like:
```c
uk_socket_evd_raddr_set(&al->evd, addr, (addr_len ? *addr_len : 0));
```

Reply:
Hi @skuenzer.
I've changed what you've suggested.
Thank you.